File: api/external_agent.py
from fastapi import FastAPI, Request
from fastapi.responses import JSONResponse
import openai
import os

app = FastAPI()

openai.api_key = os.environ["OPENAI_API_KEY"]

@app.post("/api/external_agent")
async def external_agent(request: Request):
    data = await request.json()
    query = data.get("query", "No query provided.")

    system_prompt = """
You are a general-purpose AI assistant with access to your full pretraining and general world knowledge.
Feel free to draw from your internal model, factual understanding, or past training data.
If the user is asking for something current (e.g., “latest” or “2025” or “this week”), answer to the best of your ability with a disclaimer if unsure.

This is the only agent in the system allowed to hallucinate, generalize, or reason based on your own training. Act smart and helpful.
"""

    messages = [
        {"role": "system", "content": system_prompt},
        {"role": "user", "content": query}
    ]

    response = openai.ChatCompletion.create(
        model="gpt-4",
        messages=messages
    )

    return JSONResponse({"result": response["choices"][0]["message"]["content"]})

File: api/external_agent.js
const OpenAI = require("openai");

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY
});

export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(405).json({ error: "Method not allowed" });
  }

  const { query } = req.body;

  if (!query) {
    return res.status(400).json({ error: "Missing 'query' in request body" });
  }

  try {
    const completion = await openai.chat.completions.create({
      model: "gpt-4",
      messages: [
        {
          role: "system",
          content: `
You are a general-purpose AI assistant with full access to your pretraining knowledge.
You are allowed to make informed estimates, provide general facts, and answer questions based on your internal model.

You may use past data and general world knowledge to answer questions.
Only disclaim if the request is extremely time-sensitive or if you're completely unsure.
          `
        },
        {
          role: "user",
          content: query
        }
      ],
      temperature: 0.7
    });

    const result = completion.choices[0].message.content;
    return res.status(200).json({ result });
  } catch (error) {
    console.error("OpenAI API Error:", error);
    return res.status(500).json({ error: "Error calling external agent." });
  }
}
